src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -351,25 +366,40 @@ private void EventLoop()
         private void RequestEventLoopShutdown()
         {
             //
-            // Write to the pipe, which will wake up the event loop and cause it to exit.
+            // Write to the pipe or eventfd, which will wake up the event loop and cause it to exit.
+            // (Need to write 8 bytes in case we're on a system where eventfd is supported.)
             //
-            byte b = 1;
-            int bytesWritten = Interop.Sys.Write(_shutdownWritePipe, &b, 1);
-            if (bytesWritten != 1)
+            byte[] wakeThreadUp = {0, 0, 0, 0, 0, 0, 0, 1};
+            fixed (byte *b = &wakeThreadUp[0])
             {
-                throw new InternalException(bytesWritten);
+                int bytesWritten = Interop.Sys.Write(_shutdownWritePipe, b, 8);
+                if (bytesWritten != 8)
+                {
+                    throw new InternalException(bytesWritten);
```

Comment:
This looks suspect.  In general `write` isn't guaranteed to write everything you pass in, which is why it returns how many bytes it wrote.  When we were writing 1 byte, this was fine, because it won't return 0 unless there was an error, but now that it's being called with 8, it could theoretically successfully return 1 through 7.  Does something prevent that from being the case when the fd is a pipe or an eventfd, on all platforms we target now and in the future?

Reply:
I'm aware of that.  However, for eventfd, this is always fine (it's always ready to accept a 64-bit int).  When using pipes, it's *very* unlikely that the buffer wouldn't have at least 8 bytes free, especially since these pipes are used only during shutdown and this is the only thing that's ever written to them.  So this is fine.

Reply:
> very unlikely

I don't think "very unlikely" is sufficient for code that we need to be extremely robust.

Reply:
Pipe buffers are at least a page in all systems we care about.  It's not very unlikely, it's *very* unlikely.

But, yeah, the proposed change is fine. Even if the assert is removed in release builds, this would fine: the amount of bytes written here (on pipes, where short writes are possible) doesn't really matter as it's only used to wake the thread blocked on poll().